apps/src/music/views/MusicView.jsx

```diff
@@ -197,7 +205,7 @@ class UnconnectedMusicView extends React.Component {
 
     const codeChanged = this.compileSong();
     if (codeChanged) {
-      this.executeCompiledSong();
+      this.executeCompiledSong({executingForPlay: false});
```

Comment:
Want to make sure I understand this - what happens if blocks are changed mid-playback, and there is a random block in the future? Would it still show only the chosen block?

Reply:
Excellent catch.  The timeline would render as though we weren't playing.  Changing this to check for whether we are currently playing seems to fix that scenario.